AnalyticsTelemetricsData/AnalyticsTelemetricsData/Managers/AnalyticsManager.swift

```diff
@@ -0,0 +1,298 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import RealmSwift
+import SwiftUI
+import AppTrackingTransparency
+import AdSupport
+
+/// An enum representing different types of events that can be logged.
+public enum AnalyticsType: String, PersistableEnum, CaseIterable, Identifiable {
+    /// To be used when tracking a screen view.
+    case screenView = "screen_view"
+    /// To be used every time the user opens the app.  This is done automatically by `Analytics Manager`.
+    case sessionStart = "session_start"
+    /// To be used when tracking a click event over a UI element.
+    case click = "click"
+    /// To be used when tracking login to any service.
+    case login = "login"
+    /// To be used when tracking any other event.
+    case event = "event"
+
+    public var id: Self { self }
+}
+/**
+ `RealmAnalytics` wraps all the functionality for logging data..
+ */
+@available(macOS 11.0, tvOS 14.0, iOS 14.0, *)
+public class AnalyticsManager {
+    // Shared
+    public static var shared = AnalyticsManager()
+
+    private var appId = "<ANALYTICS_APP_ID>"
+    private var apiKey = "<ANALYTICS_API_KEY>"
+
+    // Event queues
+    private var eventQueue: [AnalyticsEvent] = []
+
+    private var currentSession: AnalyticsSession?
+
+    private init() {}
+
+    /**
+     Configures your App to be used with this manager, without this step we cannot log any events.
+     */
+    public func configure() async {
+        do {
+            try await self.setUpApp()
+        } catch {
+            print("Analytics Log Error: \(error)")
+        }
+    }
+
+    /**
+     Logs an event which will dispatched to be synced.
+
+     - parameter type: The event type `AnalyticsType` you want to track.
+     - parameter label: The label that uniquely identifies you event, could be the name of the screen or the name
+                        of the event you want to track.
+     - parameter info: Extra info you want to add to your event, this info can help to categorise or provide specific
+                       info which can be used later for reporting.
+     */
+    public func logEvent(_ type: AnalyticsType, label: String, info: [String: String]? = nil) {
+        let event = AnalyticsEvent()
+        event._id = ObjectId.generate()
+        event.type = type
+        event.label = label
+        let map = Map<String, String>()
+        info?.forEach { map[$0.key]  = $0.value }
+        event.info = map
+        event.date = Date()
+        do {
+            try logEvent(event)
+        } catch {
+            print("Analytics Log Error: \(error)")
+        }
+    }
+}
+
+enum AnalyticsError: Error {
```

Comment:
mmm in every App when we disable tracking there is no error showing, because in this case the user knows that it is disabled and expects no tracking in this case. I added an error message in this case